test/unittests/cxx23_utility_test.cpp

```diff
@@ -0,0 +1,41 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2022 The Fizzy Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "cxx23/utility.hpp"
+#include "stack.hpp"
+#include <gtest/gtest.h>
+#include <array>
+#include <vector>
+
+using namespace fizzy;
+using namespace testing;
+
+TEST(cxx23_utility, to_underlying)
+{
+    enum class A
+    {
+        A = 0xff
+    };
+    EXPECT_EQ(to_underlying(A::A), 0xff);
```

Comment:
Add `static_assert(std::is_same_v<decltype(to_underlying(A::A)), int>);

Reply:
Specs said it is *at least* `int`. So this may fail on some platforms?

Reply:
It will be bigger than `int` only if some values don't fit `int`

Reply:
For scoped enumerations (`enum class`) the implicit underlying type is `int`. In case value do not fit, this is compile error. https://godbolt.org/z/1x8ffaadW.

Reply:
I just wanted to know if this returns the same type as in the enum declaration. If you want to be explicit, you can define it as `enum class A : int`.